src/compiler/factory.ts

```diff
@@ -2292,15 +2292,26 @@ namespace ts {
     /**
      * Sets a custom text range to use when emitting source maps.
      */
-    export function setSourceMapRange<T extends Node>(node: T, range: TextRange | undefined) {
+    export function setSourceMapRange<T extends Node>(node: T, range: SourceMapRange | undefined) {
         getOrCreateEmitNode(node).sourceMapRange = range;
         return node;
     }
 
     /**
+     * Create an external source map source file reference
+     */
+    export function createSourceMapSource(fileName: string, text: string, skipTrivia?: (pos: number) => number) {

```

Comment:
sourcemap.ts calls `getLineAndCharacterOfPosition()` from `scanner.ts`  with a `SourceMapSource` which then requires `SourceFileSource` to be `SourceFileLike`.

I don't want to infect `scanner.ts` with `SourceMapSource` as `SourceFileLIke` is intended to be the minimal specification of a content and cached lines. The fact it is extended to include `getLineAndCharacterOfPosition()` in `services/types.ts` seems troubling and unnecessary.